src/main/java/com/rollbar/android/Notifier.java

```diff
@@ -279,25 +281,35 @@ public void onFailure(HttpResponse response) {
                 Log.e(Rollbar.TAG, "There was a problem reporting to Rollbar.");
                 Log.e(Rollbar.TAG, "Response: " + response);
 
-                if (file == null) {
-                    if (!response.hasStatusCode()) {
+                // if there's no response code or the response code is not a 2xx or 4xx
+                //    write a file if one doesn't exist already and re-schedule
+                // TODO(eric): make this conditional check one function
+                if (!response.hasStatusCode() || !(response.has2XXStatusCode() || response.has4XXStatusCode())) {
+                    Log.d(Rollbar.TAG, "Failed response: write new file or use existing");
+                    if (file == null) {
+                        Log.d(Rollbar.TAG, "Failed response: writing a new file.");
                         writeItems(items);
                     }
+
+                    Log.d(Rollbar.TAG, "Failed response: re-scheduling.");
+                    scheduleItemFileHandler(itemScheduleDelay);
+                // otherwise, delete the file if it exists
                 } else {
-                    // Give up if there is a server error
-                    if (response.hasStatusCode()) {
+                    Log.d(Rollbar.TAG, "Failed response: don't write new file and delete if it exists");
```

Comment:
As far as the files go I believe they can contain multiple items but I haven't seen that happen testing yet. You'd probably have to fire off quite a lot of errors at once for that to happen.